hl/src/H5LT.c

```diff
@@ -781,17 +813,30 @@ H5LTopen_file_image(void *buf_ptr, size_t buf_size, unsigned flags)
         udata->vfd_image_size  = 0;
         udata->vfd_ref_count   = 0;
         udata->flags           = flags;
-        udata->ref_count       = 1; /* corresponding to the first FAPL */
+
+        /*
+         * Initially, the FAPL created in this function holds a reference
+         * to the udata structure allocated above. The call to
+         * H5Pset_file_image_callbacks below will associate the udata
+         * structure with the FAPL and cause it to hold another reference
+         * in preparation for transfer of ownership to the file driver.
+         * Once the file has been opened with this FAPL and the FAPL is
+         * closed, the reference held by the FAPL is released and ownership
+         * is transferred to the file driver. The udata structure will then
+         * be freed when the image_free callback is made and the file driver
+         * releases its reference to the structure.
```

Comment:
So this reference added here (the "1") is really always the file driver's reference right? After the call to H5Pset_file_image_callbacks the rc is 2, one for the FAPL one for the FD right? Sorry if this seems pedantic but I think it's important to always explicitly who owns a reference.

Reply:
Yes, wording this is a bit tricky because in actuality no one owns that reference to the udata structure initially. The call to `H5Pset_file_image_callbacks` comes first and associates the udata structure with the FAPL, at which point the RC is 2 and yet still no one owns that initial reference. It isn't until the H5Fopen call that the FD actually owns the reference, which isn't a particularly great design since it makes cleaning things up tricky if `H5LTopen_file_image` fails. I'll reword a bit to make this explicit.